packages/grafana-data/src/transformations/transformers/labelsToFields.test.ts

```diff
@@ -1,232 +1,104 @@
-        { name: 'temp', type: FieldType.number, values: [-1], labels: { location: 'outside', sky: 'cloudy' } },
-      ],
-    });
-
-    const result = transformDataFrame([cfg], [oneValueDifferentLabelsA, oneValueDifferentLabelsB]);
-    const expected: Field[] = [
-      { name: 'time', type: FieldType.time, values: new ArrayVector([1000, 2000]), config: {} },
-      { name: 'location', type: FieldType.string, values: new ArrayVector(['inside', 'outside']), config: {} },
-      { name: 'feelsLike', type: FieldType.string, values: new ArrayVector(['ok', null]), config: {} },
-      { name: 'sky', type: FieldType.string, values: new ArrayVector([null, 'cloudy']), config: {} },
-      { name: 'temp', type: FieldType.number, values: new ArrayVector([1, -1]), config: {} },
-    ];
-
-    expect(result[0].fields).toEqual(expected);
-  });
-
-  it('data frames with same timestamp and different labels', () => {
-    const cfg: DataTransformerConfig<LabelsToFieldsOptions> = {
-      id: DataTransformerID.labelsToFields,
-      options: {},
-    };
-
-    const oneValueDifferentLabelsA = toDataFrame({
-      name: 'A',
-      fields: [
-        { name: 'time', type: FieldType.time, values: [1000, 2000] },
-        { name: 'temp', type: FieldType.number, values: [1, 2], labels: { location: 'inside', feelsLike: 'ok' } },
+        { name: 'temp', type: FieldType.number, values: [-1], labels: { location: 'outside' } },
       ],
     });
 
-    const oneValueDifferentLabelsB = toDataFrame({
-      name: 'B',
-      fields: [
-        { name: 'time', type: FieldType.time, values: [1000, 2000] },
-        { name: 'temp', type: FieldType.number, values: [-1, -2], labels: { location: 'outside', sky: 'cloudy' } },
-      ],
-    });
+    const result = toDataFrameDTO(transformDataFrame([cfg], [oneValueOneLabelA, oneValueOneLabelB])[0]);
 
-    const result = transformDataFrame([cfg], [oneValueDifferentLabelsA, oneValueDifferentLabelsB]);
-    const expected: Field[] = [
-      { name: 'time', type: FieldType.time, values: new ArrayVector([1000, 1000, 2000, 2000]), config: {} },
-      {
-        name: 'location',
-        type: FieldType.string,
-        values: new ArrayVector(['inside', 'outside', 'inside', 'outside']),
-        config: {},
-      },
-      { name: 'feelsLike', type: FieldType.string, values: new ArrayVector(['ok', null, 'ok', null]), config: {} },
-      { name: 'sky', type: FieldType.string, values: new ArrayVector([null, 'cloudy', null, 'cloudy']), config: {} },
-      { name: 'temp', type: FieldType.number, values: new ArrayVector([1, -1, 2, -2]), config: {} },
+    const expected: FieldDTO[] = [
+      { name: 'time', type: FieldType.time, values: [1000, 2000], config: {} },
+      { name: 'location', type: FieldType.string, values: ['inside', 'outside'], config: {} },
+      { name: 'temp', type: FieldType.number, values: [1, -1], config: {} },
     ];
 
-    expect(result[0].fields).toEqual(expected);
```

Comment:
Are these tests covered by the `merge` transformer? Those tests tested data frames with different labels.

Reply:
yes, it has tests with data frames with different fields